_articles/aws_sts_storage_locations.md

```diff
@@ -0,0 +1,178 @@
+---
+title: "AWS Security Token Service Storage Locations"
+layout: article
+excerpt: Follow these steps to set up STS Storage Locations and use them to access your S3 data directly.
+category: admin-and-settings
+---
+
+Using AWS Security Token Service (STS), Synapse can securely grant you temporary AWS credentials with access to data directly in S3. This can be useful if you want to:
+
+* Transfer data in bulk
+* Allow your compute cluster to read S3 objects using the S3 APIs
+
+All of which you can now do with minimal overhead from Synapse.
+
+{% include note.html content="You can only create STS Storage Locations on an empty folder. This is to ensure consistency between the STS Storage Location and the Synapse folder." %}
+
+## Synapse-Managed STS Storage Locations
+
+You can create an STS Storage Location using Synapse storage. Synapse will store the files for you, but you can still get temporary S3 credentials, scoped to just the files and folders within your STS Storage Location.
+
+To set up the STS Storage Location using Synapse storage, first make sure you have an empty Synapse folder, then run the following code:
+
+##### Python
+
+```python
+# Set storage location
+import synapseclient
+import json
+syn = synapseclient.login()
+FOLDER = 'syn12345'
+
+destination = {'uploadType':'S3',
+               'stsEnabled':True,
+               'concreteType':'org.sagebionetworks.repo.model.project.S3StorageLocationSetting'}
+destination = syn.restPOST('/storageLocation', body=json.dumps(destination))
+
+project_destination ={'concreteType': 'org.sagebionetworks.repo.model.project.UploadDestinationListSetting',
+                      'settingsType': 'upload'}
+project_destination['locations'] = [destination['storageLocationId']]
+project_destination['projectId'] = FOLDER
+
+project_destination = syn.restPOST('/projectSettings', body = json.dumps(project_destination))
```

Comment:
What is the name of the bucket created?  You will need that to download files via s3

Reply:
Is it returned in the POST call to `/storageLocation`? Looks like a BucketStorageLocationSetting has the bucket name but I'm not sure if that's the implementation returned in this case.

Reply:
It's not. There is a get `/storageLocation` but currently the s3 bucket name isn't exposed

Reply:
Edit:  There is a way to get it by just doing a `syn.get(..., downloadFile=False)`

Reply:
I believe (but need to confirm for sure) that you can only get the File Handle back if you're the File Handle owner. Which means it will work for your own File Handles, but not for File Handles that are shared with you.

I've filed https://sagebionetworks.jira.com/browse/PLFM-6226 to make this improvement.

Reply:
@DwayneJengSage Can you confirm that the destination['baseKey'] is the root of the scope of the bucket to which the STS token allows access?

Reply:
Yes, baseKey is the root folder to which the STS token allows access. It's an S3 prefix, so it doesn't contain the bucket name.